airflow_exporter/prometheus_exporter.py

```diff
@@ -103,19 +103,21 @@ def collect(self):
         # Task metrics
         # Each *MetricFamily generates two lines of comments in /metrics, try to minimize noise 
         # by creating new group for each dag
-        task_info = get_task_state_info()
-        for dag_id, tasks in itertools.groupby(task_info, lambda x: x.dag_id):
-            k, v = get_dag_labels(dag_id)
-
-            t_state = GaugeMetricFamily(
-                'airflow_task_status',
-                'Shows the number of task starts with this status',
-                labels=['dag_id', 'task_id', 'owner', 'status', 'hostname'] + k
-            )
-            for task in tasks:
-                t_state.add_metric([task.dag_id, task.task_id, task.owners, task.state or 'none', task.hostname or 'none'] + v, task.value)
+        
+# Task metrics to be removed as we have more than 20,000 tasks on production which is causing service timing out.         
+#         task_info = get_task_state_info()
+#         for dag_id, tasks in itertools.groupby(task_info, lambda x: x.dag_id):
+#             k, v = get_dag_labels(dag_id)
+
+#             t_state = GaugeMetricFamily(
+#                 'airflow_task_status',
+#                 'Shows the number of task starts with this status',
+#                 labels=['dag_id', 'task_id', 'owner', 'status', 'hostname'] + k
+#             )
+#             for task in tasks:
+#                 t_state.add_metric([task.dag_id, task.task_id, task.owners, task.state or 'none', task.hostname or 'none'] + v, task.value)
             
-            yield t_state
```

Comment:
You can just delete code. Thats why we have versioning.

Reply:
Removed.